agent/stats/queue.go

```diff
@@ -186,6 +190,70 @@ func (queue *Queue) GetStorageStatsSet() (*ecstcs.StorageStatsSet, error) {
 	return storageStatsSet, errOut
 }
 
+// GetRestartStatsSet gets the stats set for container restarts
+func (queue *Queue) GetRestartStatsSet() (*ecstcs.RestartStatsSet, error) {
+	return queue.getRestartStatsSet(getRestartCount)
```

Comment:
do we need a read lock in this method since you're computing what the firstStat and lastStat is?

Reply:
We get a write lock inside getRestartStatsSet. I think we probably could get by with a read lock but all of the existing `get*StatsSet` functions use a write lock so I just followed that pattern.

Reply:
I saw that too! there's this `GetLastNetworkStatPerSec` that has a read lock. I wonder why that's the only one with a read lock... 🤔 

https://github.com/aws/amazon-ecs-agent/blob/7884674a94ce88281e44a25e60fc89a64c293f8e/agent/stats/queue.go#L156-L158